components/LessonsChallenges.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react'
+import '../scss/lessonsChallenges.scss'
+
+type ChallengeStatusProps = {
+  challengesData: Challenge[]
+}
+
+type LessonChallengeProps = {
+  lessons: LessonChallenge[]
+}
+
+type LessonChallenge = {
+  order: number
+  title: string
+  challenges: Challenge[]
+}
+
+type Challenge = {
+  challengeNumber: number
+  challengeStatus?: string
+}
+
+export const ChallengeStatus: React.FC<ChallengeStatusProps> = props => {
+  const displayChallengeStatus = props.challengesData.map(
+    (eachChallenge, challengeId) => {
+      let challengeStatus = 'default_challenge_status'
+      if (eachChallenge.challengeStatus === 'passed') {
+        challengeStatus = 'passed_challenge_status'
+      }
+      if (eachChallenge.challengeStatus === 'wrong') {
+        challengeStatus = 'unapproved_challenge_status'
+      }
+      if (eachChallenge.challengeStatus === 'pending') {
+        challengeStatus = 'pending_challenge_status'
+      }
+      return (
+        <span
+          key={challengeId}
+          className={`challenge_status ${challengeStatus}`}
+        ></span>
+      )
+    }
+  )
+  return <div className="challenges_container">{displayChallengeStatus}</div>
+}
+
+const LessonsChallenges: React.FC<LessonChallengeProps> = props => {
+  let filterPassedChallenges = []
```

Comment:
Can you just declare this variable as a const on line 50? I don’t think you need to reassign it multiple times and closure will allow for this variable to be persistent per lesson

Reply:
> Can you just declare this variable as a const on line 50? I don’t think you need to reassign it multiple times and closure will allow for this variable to be persistent per lesson

that makes sense, thanks for the explanation.